File: examples/Android/app/src/main/java/dslplatform/com/androidexample/DSL.java
package dslplatform.com.androidexample;

import com.dslplatform.json.DslJson;

public abstract class DSL {
    private static DslJson<Object> json;
    public static DslJson<Object> JSON() {
        if (json == null) {
            //during initialization ServiceLoader.load should pick up services registered into META-INF/services
            //this doesn't really work on Android so DslJson will fallback to default generated class name
            //"dsl_json.json.ExternalSerialization" and try to initialize it manually
            json = new DslJson<>();
        }
        return json;
    }
}

File: examples/AndroidKotlin/app/src/main/java/com/dslplatform/androidkotlin/DSL.kt
package com.dslplatform.androidkotlin

import com.dslplatform.json.DslJson
import com.dslplatform.json.runtime.Settings

object DSL {
    private var json: DslJson<Any>? = null
    fun JSON(): DslJson<Any> {
        var tmp = json
        if (tmp != null) return tmp
        //during initialization ServiceLoader.load should pick up services registered into META-INF/services
        //this doesn't really work on Android so DslJson will fallback to default generated class name
        //"dsl_json_Annotation_Processor_External_Serialization" and try to initialize it manually
        tmp = DslJson(Settings.withRuntime<Any>().includeServiceLoader().allowArrayFormat(true))
        json = tmp
        return tmp
    }
}